src/dataframe/dataframe.jl

```diff
@@ -502,34 +516,67 @@ end
         throw(BoundsError(df, (row_inds, col_inds)))
     end
     selected_columns = index(df)[col_inds]
-    # Computing integer indices once for all columns is faster
-    selected_rows = T === Bool ? findall(row_inds) : row_inds
-    new_columns = AbstractVector[dv[selected_rows] for dv in _columns(df)[selected_columns]]
-    return DataFrame(new_columns, Index(_names(df)[selected_columns]), copycols=false)
+
+    u = _names(df)[selected_columns]
+    lookup = Dict{Symbol, Int}(zip(u, 1:length(u)))
+    idx = Index(lookup, u)
```

Comment:
Isn't `idx = Index(u)` equivalent to the new code?

Reply:
no - it is not. `Index(u)` would check if `u` does not contain duplicates. And we already do this check in line `selected_columns = index(df)[col_inds]` above so this is duplicating the work (I made this change to improve benchmarks).

Reply:
OK. Can you add a comment then? It could almost make sense to add a `checkduplicates` keyword argument to `Index`.

Reply:
added